core/coreapi/dag_test.go

```diff
@@ -0,0 +1,89 @@
+package coreapi_test
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/ipfs/go-ipfs/core/coreapi"
+	"github.com/ipfs/go-ipfs/path"
+)
+
+var (
+	treeExpected = map[string]struct{}{
+		"a":   {},
+		"b":   {},
+		"c":   {},
+		"c/d": {},
+		"c/e": {},
+	}
+)
+
+func TestPut(t *testing.T) {
+	ctx := context.Background()
+	_, api, err := makeAPI(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, err = api.Dag().Put(ctx, strings.NewReader(`"Hello"`), "json", nil)
```

Comment:
I would check the Cid of the created object. Just one more additional protection against us breaking things in the future.